### PR TITLE
Handle when all remote node are unavailalble.

### PR DESCRIPTION
**Pull request by ipdae**
image

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -330,6 +331,9 @@ export const TRANSIFEX_TOKEN = "1/9ac6d0a1efcda679e72e470221e71f4b0497f7ab";
 export async function initializeNode(): Promise<NodeInfo> {
   console.log("config initialize called");
   const nodeList = await NodeList();
+  if (nodeList.length < 1) {
+    throw new InitializeNodeFailedError();
+  }
   nodeList.sort((a, b) => {
     return a.clientCount - b.clientCount;
   });
```
Comment (BasixKOR) [inline src/config.ts:339]: Not really related to this PR per se, but do we need to sort this list if we're going to pick randomly?

Comment (ipdae) [reply src/config.ts:338]: yes, this is temporary loadbalance in client side before rpc node support alb(or nlb).

File: src/main/main.ts [hunk 2]
```diff
@@ -205,7 +205,17 @@ async function initializeApp() {
...
-    remoteNode = await initializeNode();
+    try {
+      remoteNode = await initializeNode();
+    } catch (e) {
+      await dialog.showMessageBox(win!, {
+        message: "Failed to connect remote node. please restart launcher.",
+        type: "error",
+      });
+
+      win.webContents.send("go to error page", "initialize-node-failed");
```
Comment (BasixKOR) [inline src/main/main.ts:216]: This line doesn't really do anything since it will be exited immediately.

File: src/main/main.ts [hunk 3]
```diff
@@ -205,7 +205,17 @@ async function initializeApp() {
     if (app.commandLine.hasSwitch("v2")) win = await createV2Window();
     else win = await createWindow();
     createTray(path.join(app.getAppPath(), logoImage));
-    remoteNode = await initializeNode();
+    try {
+      remoteNode = await initializeNode();
+    } catch (e) {
```
Comment (longfin) [inline src/main/main.ts:210]: How about adding a log related to `e`? (i.e. `console.error(e)`)

**Review by longfin — APPROVED** [review]

File: src/main/main.ts (hunk 3 above)
Comment (BasixKOR) [reply src/main/main.ts:210]: Nice catch. It would be better if the thrown error on `config.ts` included more information about the error.

**Review by moreal — APPROVED** [review]

**Review by area363 — APPROVED** [review]
👍

**Review by longfin — APPROVED** [review]

**Pull request merged**